core/zoe/isOfferSafe.js

```diff
@@ -0,0 +1,105 @@
+import { bothTrue } from './utils';
+import { insist } from '../../util/insist';
+
+/**
+ * `isOfferSafeForPlayer` checks offer-safety for a single player.
+ *
+ * Note: This implementation checks whether we refund for all rules or
+ * return winnings for all rules. It does not allow some refunds and
+ * some winnings, which is what would happen if you checked the rules
+ * independently.
+ *
+ * @param  {array} assaysPerIssuer - an array of assays ordered in the
```

Comment:
Very good point. I was originally thinking that offer safety included an issuer check (which operating on quantities using strategies doesn't give you) but I think we can safely assume that that is done elsewhere in Zoe, when the payments are made using the array of issuers. 

I was able to change to quantities/strategies, so that change is in, at least for this PR!